test/Sprite/TilesetLoader.test.cpp

```diff
@@ -27,14 +27,16 @@ TEST(TilesetLoader, ReadTileset)
 	Stream::DynamicMemoryWriter writer;
 	tileset.WriteIndexed(writer);
 	
-	auto newTileset = Tileset::ReadTileset(writer.GetReader());
+	auto reader = writer.GetReader();
+	auto newTileset = Tileset::ReadTileset(reader);
```

Comment:
This is somewhat surprising. I feel like the original syntax should be made possible. Maybe an rvalue overload that delegates to an lvalue overload?

A stream is necessarily modifiable, as the stream position is mutable state, so a `const` reference is not possible. But only a const lvalue reference can bind to an rvalue. To accept a temporary, we need a method that accepts an rvalue. If we had only an rvalue overload, we'd need to cast or `std::move` a named variable to pass an lvalue. Seems like we would need both overloads here to support all typical calling cases.

Reply:
I will have to research more on lvalule/rvalue to implement your suggestions as my understanding is pretty weak of the concept. 

I agree with the end state of being able to make the call based on my original syntax. MSVC allowed for the previous behaviour although I have seen this sort of call fail on Linux based compilers in the past. I think MSVC is being a bit more permissive than specified by the C++ standard by allowing it to compile in the first place.

Reply:
Oh, right. I remember this now. C++ is maddening trying to remember all the nuances of the language.

We implemented this in the Archive code for writing volumes / clm files for the same reason.